src/io/iter_prefetcher.h

```diff
@@ -79,17 +79,22 @@ class PrefetcherIter : public IIterator<DataBatch> {
           (*dptr)->data.resize(batch.data.size());
           (*dptr)->index.resize(batch.batch_size);
           for (size_t i = 0; i < batch.data.size(); ++i) {
+            auto dtype = param_.dtype != mshadow::default_type_flag
```

Comment:
With this change, param_.dtype still takes effect for compatibility.  Prefetcher's dtype should be the same as its source unless it is specified in the param.